src/object/obj_tx.c

```diff
@@ -875,6 +875,21 @@ dc_tx_local_close(daos_handle_t th)
 	return rc;
 }
 
+static int
+dc_tx_query_key(struct dc_tx *tx, uint64_t flags,
+		daos_key_t *dkey, daos_key_t *akey)
+{
+	tse_task_t	*task;
+	int		 rc;
+
+	rc = dc_obj_query_key_task_create(tx->tx_coh, dc_tx_ptr2hdl(tx), flags,
+					  dkey, akey, NULL, NULL, NULL, &task);
+	if (rc == 0)
+		rc = dc_task_schedule(task, true);
```

Comment:
Does this mean we'll block the application until this key query task terminates?

Reply:
Yes, we have to wait the key query task, because the application may depend on the conditional operation's return value to make decision for next action.

Reply:
Yes. To be exact, if the user calls such an update function with an event, then the _function_ should return, while the _operation_ waits for the key query task.

Reply:
So that breaks the async behavior that would be expected even with a distributed transaction.
why can't the query task be created and inserted into the schedule asynchronously, and the checks be done in the completion callback?